.changeset/long-apes-admire/changes.md

```diff
@@ -0,0 +1,38 @@
+TypeScript types have been restructured. These changes:
+
+- Reduce build times when using emotion
+- In many cases remove the need for manually specifying generic parameters for your emotion components.
+
+If you encounter build issues after upgrade, try removing any manually specified generic types and let them be inferred. Otherwise refer to the breaking changes list below.
+
+## Improvements
+
+- useTheme added to EmotionTheming interface and can now create your own closed variation of withTheme. More information in the docs under the theming section.
+- Union types as props are better supported and should be inferred properly
+- Build times should be reduced significantly in larger projects.
+
+## Breaking changes
+
+- Function interpolation can no longer return another function directly
+- withTheme can now have the Theme type specified when calling it. For example `withTheme<MyTheme>(MyComponent)`
+
+  **Breaking change:** Generic argument changed, if you were specifying the ComponentType you will need to remove the generic parameter. Recommend following example setup in the TypeScript docs under theming section
+
+- `css` function has been restricted to prevent passing of invalid types
+- `CreateStyled` functions no longer take a second `ExtraProps` argument. Instead move it to after the create styled call. For example
+
+  `styled<typeof MyComponent, ExtraProps>(MyComponent)({})`  
+  to  
+  `styled(MyComponent)<ExtraProps>({})`
+
+- `CreateStyledComponentBase`, `CreateStyledComponentBaseThemeless` and `CreateStyledComponentBaseThemed` have been merged into `CreateStyledComponent`.
```

Comment:
I'm not entirely sure, but at least `CreateStyledComponentBaseThemeless` and `CreateStyledComponentBaseThemed` weren't meant for public usage, so we can avoid mentioning them here to avoid the unnecessary mental overhead for readers. Not sure about `CreateStyledComponentBase` - that would have to be checked.

Reply:
Honestly if you are following the docs you shouldn't really be using any of those types directly. Happy to just remove this line.

Reply:
I think those complex ones can be removed from here - as they were never intended to be used publicly.

Reply:
These have been simplfied